packages/react/src/auto/shadcn/inputs/relationships/RelatedModelOption.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import type { Option } from "../../../interfaces/AutoRelationshipInputProps.js";
+import type { ShadcnElements } from "../../elements.js";
+import { makeShadcnListMessages } from "./ShadcnListMessages.js";
+
+// TODO: This is Polaris's AutoSelection enum, but it might not be needed for shadcn.
+export declare enum AutoSelection {
+  /** Default active option is the first selected option. If no options are selected, defaults to first interactive option. */
+  FirstSelected = "FIRST_SELECTED",
+  /** Default active option is always the first interactive option. */
+  First = "FIRST",
+  /** Default to the manual selection pattern. */
+  None = "NONE",
+}
+
+export type RelatedModelOptionsProps = {
+  options: Option[];
+  records?: Record<string, any>[];
+  isLoading?: boolean;
+  errorMessage?: string;
+  checkSelected?: (id: string) => boolean;
+  onSelect: (record: Record<string, any>) => void;
+  autoSelection?: AutoSelection;
+  actions?: React.ReactNode[];
+  renderOption?: (option: Option) => React.ReactNode;
+  allowMultiple?: boolean;
+};
+
+export const makeRelatedModelOption = ({
+  CommandItem,
+  CommandList,
+  CommandEmpty,
+  CommandGroup,
+  Checkbox,
+  Label,
+}: Pick<ShadcnElements, "CommandItem" | "CommandList" | "CommandEmpty" | "CommandGroup" | "Checkbox" | "Label">) => {
+  const RelatedModelOption = React.memo(function RelatedModelOption(props: RelatedModelOptionsProps) {
+    const { checkSelected, onSelect, isLoading, errorMessage, options, records, actions } = props;
+
+    const { ListMessage, NoRecordsMessage, ShadcnSelectableOption, getErrorMessage } = React.useMemo(
+      () =>
+        makeShadcnListMessages({
+          CommandItem,
+          CommandEmpty,
+          Checkbox,
+          Label,
+        }),
+      [CommandItem, CommandEmpty, Checkbox, Label]
+    );
```

Comment:
I think this should be moved out of the inner component body and only defined once and not memoized